### PR TITLE
Adding Cloud WAN support

### PR DESCRIPTION
**Pull request by pablo19sc**
Adding Cloud WAN support:

* Creation of core_network subnets.
* Creation of Core Network's VPC attachment.
* Support for attachment acceptance for VPC attachments.

### CODE REVIEW

**Comment by pablo19sc** [comment]
Test of new example (cloud_wan)

test_cloud_wan

File: .header.md [hunk 1]
```diff
@@ -41,7 +41,7 @@ module "vpc" {
 
 ## Reserved Subnet Key Names
 
-There are 2 reserved keys for subnet key names in var.subnets corresponding to types "public" and "transit_gateway". Other custom subnet key names are valid are and those subnets will be private subnets.
+There are 3 reserved keys for subnet key names in var.subnets corresponding to types "public", "transit_gateway", and "core_network". Other custom subnet key names are valid are and those subnets will be private subnets.
```
Comment (drewmullen) [inline .header.md:44]:
```suggestion
There are 3 reserved keys for subnet key names in var.subnets corresponding to types "public", "transit_gateway", and "core_network" [(a NetworkManager CloudWAN feature)](https://docs.aws.amazon.com/vpc/latest/cloudwan/cloudwan-networks-working-with.html). Other custom subnet key names are valid are and those subnets will be private subnets.
```

**Review by drewmullen — CHANGES_REQUESTED** [review]
beautiful work. only 1 request here - lmk what you think

File: .header.md (hunk 1 above)
Comment (drewmullen) [reply .header.md:44]: my thought here is that _most people_ dont know cloudwan but the term "core network" feels very confusing.

**Review by drewmullen — APPROVED** [review]
LGTM

**Pull request merged**